### PR TITLE
Adds some more taming syntax

### PR DESCRIPTION
**Pull request by cheeezburga**
This PR adds an effect to tame an entity and a condition to check if an entity is tamed.

### CODE REVIEW

**Review by NotSoDelayed — CHANGES_REQUESTED** [review]
Add missing Skript docs

File: src/main/java/ch/njol/skript/conditions/CondIsTamed.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Tameable;
+
+@Name("Is tamed")
```
Comment (NotSoDelayed) [inline src/main/java/ch/njol/skript/conditions/CondIsTamed.java:11]:
```suggestion
@Name("Is Tamed")
```

File: src/main/java/ch/njol/skript/conditions/CondIsTamed.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+@Description("Check if a tameable entity is tamed (horse, parrot, cat, etc.).")
+@Examples({"send true if {_horse} is tamed",
+	"tame {_horse} if {_horse} is untamed"})
```
Comment (NotSoDelayed) [inline src/main/java/ch/njol/skript/conditions/CondIsTamed.java:14]:
```suggestion
@Examples({
    "send true if {_horse} is tamed",
	"tame {_horse} if {_horse} is untamed"
})
```
Although im not sure if we would use Skript's ternary expression as apart of the examples, just saying.

File: src/main/java/ch/njol/skript/effects/EffTame.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Tameable;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Tame / Untame")
+@Description("Tame a tameable entity (horse, parrot, cat, etc.).")
+@Examples({"tame {_horse}",
+	"untame {_horse}"})
```
Comment (NotSoDelayed) [inline src/main/java/ch/njol/skript/effects/EffTame.java:20]:
```suggestion
@Examples({
    "tame {_horse}",
	"untame {_horse}"
})
```

File: src/main/java/ch/njol/skript/conditions/CondIsTamed.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+		if (entity instanceof Tameable)
+			return ((Tameable) entity).isTamed();
+		return false;
+	}
+
+	@Override
+	protected String getPropertyName() {
+		return "tamed";
+	}
+}
```
Comment (NotSoDelayed) [inline src/main/java/ch/njol/skript/conditions/CondIsTamed.java:33]:
```suggestion
	}

}
```
Skript's standardised formatting

File: src/main/java/ch/njol/skript/effects/EffTame.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+	static {
+		Skript.registerEffect(EffTame.class, "([un:(un|de)]tame|domesticate) %entities%");
+	}
+
+	private boolean tame;
+	private Expression<Entity> entities;
+
+	@SuppressWarnings("unchecked")
+	@Override
```
Comment (NotSoDelayed) [inline src/main/java/ch/njol/skript/effects/EffTame.java:32]:
```suggestion
	@Override
	@SuppressWarnings("unchecked")
```

File: src/test/skript/tests/syntaxes/conditions/CondIsTamed.sk [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+	tame {_}
+	assert {_} is not tamed with "taming null should do nothing"
+
+	spawn a zombie at (spawn of world "world")
+	set {_z} to last spawned zombie
+	tame {_z}
+	assert {_z} is not tamed with "taming an invalid entity should do nothing"
+
+	kill value within {_e}
+	kill value within {_z}
```
Comment (NotSoDelayed) [inline src/test/skript/tests/syntaxes/conditions/CondIsTamed.sk:19]:
```suggestion
	kill value within {_z}

```
Skript's standardised formatting

**Review by NotSoDelayed — CHANGES_REQUESTED** [review]

File: src/test/skript/tests/syntaxes/conditions/CondIsTamed.sk (hunk 6 above)
[the remark this answers is not in the document]
Comment (cheeezburga) [reply src/test/skript/tests/syntaxes/conditions/CondIsTamed.sk]: Deleting throws an error: `value within {_e} (and {_z}) cannot be deleted`. Isn't killing fine for this anyway? The values in those variables are never gonna not be horse and zombie.

[the remark this answers is not in the document]
Comment (APickledWalrus) [reply src/test/skript/tests/syntaxes/conditions/CondIsTamed.sk:22]: Try doing `delete entity within {_e}`

File: src/main/java/ch/njol/skript/effects/EffTame.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+		for (Entity entity : entities.getArray(event)) {
+			if (entity instanceof Tameable)
+				((Tameable) entity).setTamed(tame);
+		}
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		return (tame ? "tame " : "untame ") + entities.toString(event, debug);
+	}
```
Comment (NotSoDelayed) [inline src/main/java/ch/njol/skript/effects/EffTame.java:52]:
```suggestion
	}

```
Missed one for Skript’s standardised formatting

**Review by NotSoDelayed — CHANGES_REQUESTED** [review]

**Review by NotSoDelayed — APPROVED** [review]

**Review by sovdeeth — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

File: src/main/java/ch/njol/skript/conditions/CondIsTamed.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+@Since("INSERT VERSION")
+public class CondIsTamed extends PropertyCondition<Entity> {
+
+	static {
+		register(CondIsTamed.class, "(tamed|domesticated)", "entities");
+	}
+
+	@Override
+	public boolean check(Entity entity) {
+		if (entity instanceof Tameable)
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/conditions/CondIsTamed.java:26]: `return entity instance of Tameable && ((Tameable) entity).isTamed();`

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]
Looking good pull request

**Pull request merged**